lib/ilp.js

```diff
@@ -38,8 +38,8 @@ class Ilp {
 
       stream.once('close', () => {
         this.log.info('Payment stream closed')
-        if (state.moneySent) resolve()
-        else reject(new Error('Stream closed without sending all money'))
+        if (state.moneySent) resolve({ success: true, remainingAmount: 0 })
+        else resolve({ success: false, remainingAmount: ilpAmount - state.totalSent })
       })
 
       stream.once('error', (err) => {
```

Comment:
I think this will need a change as well -- it could emit an error part way through sending, and then it will reject and we'll doublepay. We need to make `sendMoney` literally never reject for any circumstance; and the return always provides the remainingAmount and any errors encountered on the way.

Reply:
if that's the case, should we wrap basically all of this logic into a try catch as well, in case something throws unexpectedly we can resolve with the remaining amount?